### ISSUE TITLE
Allow x-faro-session-id header for faro receiver

### ISSUE THREAD

**Issue by Starefossen** [grafana/agent#5836]
### What's wrong?

With the latest version of the Grafana Faro SDK (github.com/grafana/faro-web-sdk) it is sending a `x-faro-session-id` cors request header which is currently not allowed by Grafana Agent and preflight requests fails.

Grafana Agend worked until v1.2.4 of Grafana Faro SDK.

Ref. https://github.com/grafana/faro-web-sdk/issues/386

### Steps to reproduce

* Run latest version of Grafana Agent
* Run latest version of Grafana Faro Web SDK

https://github.com/grafana/faro-web-sdk/blob/main/docs/sources/tutorials/quick-start-browser.md

### Configuration

```text
apiVersion: v1
kind: ConfigMap
data:
  agent.yaml: |
    server:
      log_level: info
    metrics:
      wal_directory: /tmp/wal
      global:
        scrape_interval: 1m
      configs:
        - name: default
          remote_write:
            - url: http://monitoring-tenant-prometheus:9090/api/v1/write
    logs:
      positions_directory: /tmp/loki-pos
      configs:
        - name: default
          scrape_configs: []
          clients:
            - url: http://loki-write:3100/loki/api/v1/push
    traces:
      # https://grafana.com/docs/tempo/latest/setup/set-up-test-app/#set-up-remote_write-to-your-tempo-cluster
      configs:
        - name: default
          remote_write:
            - endpoint: tempo-distributor:4317
              insecure: true
              protocol: grpc
          receivers:
            otlp:
              protocols:
                grpc:
                http:
                  cors:
                    max_age: 7200
                    allowed_origins:
                      - '*.**REDACTED**'
                      - '*.**REDACTED**'
                      - '*.**REDACTED**'
                      - '*.**REDACTED**
                      - '*.**REDACTED**'

    integrations:
      metrics:
        autoscrape:
          enable: true
          metrics_instance: 'default'
      app_agent_receiver_configs:
        - autoscrape:
            enable: true
            metrics_instance: 'default'
          #api_key: 'secret' # FIXME. optional, if set, client will be required to provide it via x-api-key header
          instance: 'frontend'
          logs_instance: 'default'
          traces_instance: 'default'
          server:
            host: 0.0.0.0
            port: 12347
            cors_allowed_origins:
              - '*.**REDACTED**'
              - '*.**REDACTED**'
              - '*.**REDACTED**'
              - '*.**REDACTED**'
              - '*.**REDACTED**'
          logs_labels: # labels to source or override; will be added to loki log record
            stack: faro
            app_name: ""
            kind: ""
          logs_send_timeout: 5000
          sourcemaps:
            download: true # will download source file, extract source map location,
  strategies.json: '{"default_strategy": {"param": 0.001, "type": "probabilistic"}}'
```

### Logs

![image (2)](https://github.com/grafana/agent/assets/968267/1f7d27fb-527a-4d7e-82ad-6bc4344bbbed)

```
curl 'https://**READACTED**/collect' \
  -X OPTIONS \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:120.0) Gecko/20100101 Firefox/120.0' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.5' \
  -H 'Accept-Encoding: gzip, deflate, br' \
  -H 'Access-Control-Request-Method: POST' \
  -H 'Access-Control-Request-Headers: content-type,x-faro-session-id' \
  -H 'Referer: https://**REDACTED**/' \
  -H 'Origin: https://**REDACTED**' \
  -H 'Connection: keep-alive' -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-site' \
  -H 'Pragma: no-cache' \
  -H 'Cache-Control: no-cache' \
  -H 'TE: trailers'
```

**Comment by Starefossen** [comment]
This is solved in https://github.com/grafana/agent/commit/cc7cb3716ed4106326bb064882ee235694cbf9b2 (#5835), would it be possible to get a patch release out? 🙏🏻

**Comment by rfratto** [comment]
Closed by #5835 (we close issues once PRs are merged, not when patches are released). We'll release a patch as soon as we can; there's a few other bugfixes that we'll want to include in a patch so a patch is waiting on that.
